openrtb_ext/imp_appnexus_Keywords_test.go

```diff
@@ -7,29 +7,27 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-func TestKeywordsUtilUnmarshalJSON(t *testing.T) {
+func TestKeywordsUnmarshalJSON(t *testing.T) {
 	type keywords struct {
 		Keywords Keywords `json:"keywords"`
 	}
 
-	t.Run("dynamic-json", func(t *testing.T) {
-		jsonData := []byte(`{"keywords" : { "pets": ["dog"] }}`)
-		var keywords keywords
-		assert.NoError(t, json.Unmarshal(jsonData, &keywords))
-		assert.Equal(t, "pets=dog", keywords.Keywords.String())
-	})
+	type validTest struct {
+		input    []byte
+		expected string
+	}
 
-	t.Run("json-array", func(t *testing.T) {
-		jsonData := []byte(`{"keywords" : [{"key": "foo", "value": ["bar","baz"]},{"key": "valueless"}]}`)
-		var keywords keywords
-		assert.NoError(t, json.Unmarshal(jsonData, &keywords))
-		assert.Equal(t, "foo=bar,foo=baz,valueless", keywords.Keywords.String())
-	})
+	validTests := []validTest{
+		{input: []byte(`{"keywords" : { "pets": ["dog"] }}`), expected: "pets=dog"},
+		{input: []byte(`{"keywords" : { "foo":[] }}`), expected: "foo"},
+		{input: []byte(`{"keywords" : [{"key": "foo", "value": ["bar","baz"]},{"key": "valueless"}]}`), expected: "foo=bar,foo=baz,valueless"},
+		{input: []byte(`{"keywords" : "foo=bar,foo=baz,valueless"}`), expected: "foo=bar,foo=baz,valueless"},
+		{input: []byte(`{"keywords" : ""}`), expected: ""},
```

Comment:
Please also consider adding tests cases for invalid/empty JSON like
 ```
 []byte(`{"keywords" : {}}`), []byte(`{keywords": []}`), []byte(`{keywords": [{]}`), []byte(`{"keywords" : {"}}`)
 ```

Reply:
Added invalid test cases in [825b164](https://github.com/prebid/prebid-server/pull/2649/commits/825b1640408890f3c12a4196a6114b72fb15b1e6)